### PR TITLE
CIRC-1566 Aged to lost: Closing "Lost and paid" status - ACTUAL COST

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
Purpose: Once the patron has paid for the item (or the associated fines have otherwise been taken care of - for example, if the library waives them), the loan can be closed.

The lost item fee policy will specify that either the actual cost of the item is charged or a set cost for a lost item is charged (in addition to a lost item processing fee). This story covers scenarios where the set cost is charged (automated).

Resolves: [CIRC-1566](https://issues.folio.org/browse/CIRC-1566)

### CODE REVIEW

File: src/main/java/org/folio/circulation/CirculationVerticle.java [hunk 1]
```diff
@@ -138,6 +139,7 @@ public void start(Promise<Void> startFuture) {
           startFuture.fail(result.cause());
         }
       });
+    new ExpiredActualCostProcessingResource(client).register(router);
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/CirculationVerticle.java:142]: Why is it not higher, together with other resources registrations?

File: src/main/java/org/folio/circulation/domain/ActualCostRecord.java [hunk 2]
```diff
@@ -31,4 +31,5 @@ public class ActualCostRecord {
   private String feeFineTypeId;
   private String feeFineType;
   private ZonedDateTime creationDate;
+  private ZonedDateTime expirationDate;
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/ActualCostRecord.java:34]: Is this field already added on the mod-circulation-storage side?

File: src/main/java/org/folio/circulation/domain/Loan.java [hunk 3]
```diff
@@ -298,7 +299,16 @@ public Loan withUser(User newUser) {
...
+      checkoutServicePoint, originalDueDate, previousDueDate, policies, accounts, actualCostRecord);
+  }
+
+  public Loan withActualCostRecord(ActualCostRecord actualCostRecord) {
+    return new Loan(representation, item, user, proxy, checkinServicePoint, checkoutServicePoint,
+      originalDueDate, previousDueDate, policies, accounts, actualCostRecord);
+  }
+
+  public ActualCostRecord getActualCostRecord() {
+    return actualCostRecord;
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/Loan.java:311]: Lombok getter?

File: src/test/java/api/support/fixtures/FeeFineAccountFixture.java [hunk 4]
```diff
@@ -12,6 +12,9 @@
 import api.support.builders.FeefineActionsBuilder;
 import api.support.http.ResourceClient;
 import io.vertx.core.json.JsonObject;
+import static org.folio.circulation.domain.FeeFine.LOST_ITEM_ACTUAL_COST_FEE_TYPE;
+import static org.folio.circulation.domain.FeeFine.LOST_ITEM_FEE_TYPE;
+import static org.folio.circulation.domain.FeeFine.LOST_ITEM_PROCESSING_FEE_TYPE;
```
Comment (alexanderkurash) [inline src/test/java/api/support/fixtures/FeeFineAccountFixture.java:17]: Create constants for tests? Check formatting / import order

File: src/test/java/api/support/fixtures/FeeFineAccountFixture.java [hunk 5]
```diff
@@ -131,6 +141,13 @@ public IndividualResource createManualFeeForLoan(IndividualResource loan, double
     return account;
   }
 
+  private JsonObject getLostItemFeeAccount(UUID loanId) {
+    return accountsClient.getMany(exactMatch("loanId", loanId.toString())
+        .and(exactMatch("feeFineType", "Lost item fee")))
+      .getFirst();
+  }
+
+
```
Comment (alexanderkurash) [inline src/test/java/api/support/fixtures/FeeFineAccountFixture.java:150]: remove blank line

File: src/main/java/org/folio/circulation/domain/Loan.java (hunk 3 above)
Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/circulation/domain/Loan.java]: Add getters to specific fields

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**